docs/sources/developers/plugins/error-handling.md

```diff
@@ -0,0 +1,59 @@
++++
+title = "Error handling"
+type = "docs"
++++
+
+# Error handling
+
+This guide explains how to handle errors in plugins.
+
+## Provide usable defaults
+
+Allow the user to learn your plugin in small steps. By providing a usable default configuration, the user can get started right away, and you can avoid unnecessary error messages early on.
+
+For example, by selecting the first field of a expected type, the panel can display a visualization without any user configuration. If a user explicitly selects a field we use that one, otherwise we default to the first field of type `string` that we find:
+
+```ts
+const numberField = frame.fields.find(field =>
+  options.numberFieldName ? field.name === options.numberFieldName : field.type === FieldType.number
+);
+```
+
+## Display errors messages
+
+To display an error message to the user, `throw` an `Error` with the message you want to display:
+
+```ts
+throw new Error('An error occurred.');
+```
+
+Grafana displays the error message in the top-left corner of the panel.
+
+{{< docs-imagebox img="/img/docs/panel_error.png" class="docs-image--no-shadow" max-width="850px" >}}
+
+Avoid displaying overly technical error messages to the user. If you want to let technical users to be able to report the error, consider logging the error instead.
+
+```ts
+try {
+  failingFunction();
+} catch (err) {
+  console.error(err);
+  throw new Error('Something went wrong.');
+}
+```
+
+Here are some examples of situations where you might want to display an error to the user.
+
+### Invalid query response
+
+Users have full freedom when they create data source queries for panels. If your panel plugin requires a specific format for the query response, use the panel canvas to guide the user.
+
+```ts
+if (!numberField) {
+  throw new Error('Query result is missing a number field.')
```

Comment:
Should we really end exception messages with a period? It's normally not good style.